db/indexed_docs_test.go

```diff
@@ -605,6 +605,12 @@ func TestNonUniqueCreate_IfUponIndexingExistingDocsFetcherFails_ReturnError(t *t
 	}
 }
 
+/*
+// Todo - the mock in this test would need to construct the version history properly
+// however that is kind of temporary as the prod. code needs to change in a little bit
+// to bypass the lines that this trips up on once you are happy with the way things are running
+// if it still fails then, consider removing this teset (it looks low value compared to the effort
+// required to make it work)
```

Comment:
But the suggestion is not the behaviour I want.  I want Lens migrations to always run, regardless of what value `fetcherFactory` may be. `fetcherFactory` is production code, and as such should be treated as production code.  Currently it may only be used for testing, but not even the name of the prop suggests that that there is very little preventing someone from using it for production reasons (e.g. indexed, versioned, deleted-doc fetchers).

Reply:
then maybe instead of sneakily injecting this factory we should implement a proper injection and maybe revive this task https://github.com/sourcenetwork/defradb/issues/1586

the `fetcherFactory` is private and can be accessed only from the package. So there shouldn't be any problem controlling who and when is messing with it

Reply:
I have fixed the tests.